### PR TITLE
Support float 16 based primitives

### PR DESCRIPTION
**Pull request by kballeda**

### CODE REVIEW

File: test/onemkl.jl [hunk 1]
```diff
@@ -807,5 +813,34 @@ end
                 @test C ≈ h_C
             end
         end
+        @testset "gemm!" begin
```
Comment (pengtu) [inline test/onemkl.jl:839]: It seems that we will need to add Float16, ComplexF16 to the GEMM test similar to the copy test.

Comment (kballeda) [reply test/onemkl.jl:839]: Thanks for the review, yes, it is added now.

Comment (pengtu) [reply test/onemkl.jl:839]: Looks good and thank you!

**Comment by pengtu** [comment]
Very good, thank you!

**Comment by kballeda** [comment]
image

**Pull request closed without merging**